### PR TITLE
Feature/index scan limit multiple rules

### LINKED ISSUE #2950: [Feature] index scan limit push down
from issue #2602 

Related PR:
#2905 
#2823 
#2796

### PR DESCRIPTION
**Pull request by Shylock-Hg**
Close #2950

### CODE REVIEW

File: tests/tck/features/lookup/LookUpLimit.feature [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+      | "Jason Kidd"        | "Steve Nash"         | 0       |
+      | "Klay Thompson"     | "Stephen Curry"      | 0       |
+      | "Luka Doncic"       | "Dirk Nowitzki"      | 0       |
+      | "Luka Doncic"       | "Kristaps Porzingis" | 0       |
+      | "Paul Gasol"        | "Kobe Bryant"        | 0       |
+    And the execution plan should be:
+      | id | name                | dependencies | operator info  |
+      | 4  | DataCollect         | 5            |                |
+      | 5  | Sort                | 6            |                |
+      | 6  | Project             | 7            |                |
```
Comment (yixinglu) [inline tests/tck/features/lookup/LookUpLimit.feature:103]: The limit is missing in the graph layer

Comment (bright-starry-sky) [reply tests/tck/features/lookup/LookUpLimit.feature:103]: > The limit is missing in the graph layer

I think ok, limit node should not appear in the graph layer for this query. WDYT?
`lookup on tag where .. limit | limit` should appear both storage and graph layer.

**Comment by bright-starry-sky** [comment]
Generally looks good for me. PTAL @yixinglu @CPWstatic

File: tests/tck/features/lookup/LookUpLimit.feature (hunk 1 above)
Comment (Shylock-Hg) [reply tests/tck/features/lookup/LookUpLimit.feature:103]: The limit clause of lookup is different with individual limit sentence. It means limit by partition.

**Review by critical27 — APPROVED** [review]
LGTM, quick one

**Pull request merged**